safe/storage/safe_layer.py

```diff
@@ -0,0 +1,128 @@
+# coding=utf-8
+"""
+InaSAFE Disaster risk assessment tool developed by AusAid -
+**Impact Function Base Class**
+
+Contact : ole.moller.nielsen@gmail.com
+
+.. note:: This program is free software; you can redistribute it and/or modify
+     it under the terms of the GNU General Public License as published by
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+"""
+
+__author__ = 'imajimatika@gmail.com'
+__revision__ = '$Format:%H$'
+__filename = 'safe_layer'
+__date__ = '7/14/15'
+__copyright__ = ('Copyright 2012, Australia Indonesia Facility for '
+                 'Disaster Reduction')
+
+from qgis.core import QgsVectorLayer, QgsMapLayer
+from safe.storage.layer import Layer
+from safe.common.exceptions import KeywordNotFoundError
+from safe.utilities.keyword_io import KeywordIO
+
+
+class SafeLayer:
+    """Wrapper for QgsMapLayer and safe.storage.layer.
+    """
+
+    def __init__(self, layer, name=None):
+        """Init function.
+        :param layer: A layer. Can be QgsMapLayer or safe.storage.layer.
+        :type layer: QgsMapLayer, Layer
+
+        :param name: A layer's name
+        :type name: Basestring or None
+        """
+        self._layer = layer
+        if isinstance(layer, Layer):
+            self._keywords = layer.keywords
+        elif isinstance(layer, QgsMapLayer):
+            keyword_io = KeywordIO()
+            self._keywords = keyword_io.read_keywords(layer)
+        else:
```

Comment:
Maybe we should raise an exception here? Or do we have a use case for allowing initialisation with no layer?